### PR TITLE
Build Flux Architecture

### PR DESCRIPTION
**Pull request by ongmin**
:fire: @caalberts Do not accept this pullrequest :fire:

Hello @sarmadsangi! 
thanks for agreeing to give us some pointers. 

- So the entry point for Actions is InitializeActions.
- The main Action file is called Actions
- There is only 1 Store file, called Store
- SearchAPI is working, but not yet ChannelAPI and PlaylistAPI (I have commented them out in InitializeAction)

**Problem:** Initialising is fetching data so that is great, but I'm struggling to configure the onChange, componentDidMount, etc in **'./app/components/channel/searchBar'**. Could you kindly offer some pointers on the Flux files? I'm essentially trying to emulate (with Flux) the same exercise you did with us.

Thank you very much!

cc: @caalberts @cbas

### CODE REVIEW

File: app/components/search/app.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
+<<<<<<< HEAD
```
Comment (cbas) [inline app/components/search/app.js]: Whoa... suddenly unresolved merge conflict!

Comment (ongmin) [inline app/components/search/app.js]: Not for merging!

**Comment by caalberts** [comment]
@ongmin i've merged this branch into `dev`. refer to PR #25

**Comment by caalberts** [comment]
closing this. already merged in #25

**Pull request closed without merging**